### PR TITLE
Add SRT restful API

### PR DESCRIPTION
**Pull request by qwu16**

### CODE REVIEW

File: scripts/installCommonDeps.sh [hunk 1]
```diff
@@ -20,6 +20,17 @@ check_proxy(){
   fi
 }
 
+install_srt(){
+  local SRC_URL="https://github.com/Haivision/srt.git"
+  local SRC_DIR="srt"
+  mkdir -p ${LIB_DIR}
+  pushd ${LIB_DIR}
+  git clone ${SRC_URL}
```
Comment (starwarfan) [inline scripts/installCommonDeps.sh:28]: Better to use a certain version of SRT instead of the latest master

File: scripts/release/init-all.sh [hunk 2]
```diff
@@ -39,6 +39,7 @@ init_software()
     ${ROOT}/management_api/init.sh
     ${ROOT}/webrtc_agent/install_deps.sh
     ${ROOT}/video_agent/install_deps.sh
+    ${ROOT}/streaming_agent/install_ffmpeg.sh --enable-srt
```
Comment (starwarfan) [inline scripts/release/init-all.sh:42]: Is the '--enable-srt' option mandatory in init script?

File: scripts/release/install_ffmpeg.sh [hunk 3]
```diff
@@ -57,16 +78,40 @@ install_build_deps() {
...
+  local SRC_URL="https://github.com/Haivision/srt.git"
+  local SRC_DIR="srt"
+  local DIR="srt"
+  local PREFIX_DIR="${this}/ffmpeg-install"
+
+  echo "Prefix dir "${PREFIX_DIR}
+  pushd ${this} >/dev/null
+  rm -rf ${DIR}
+  ls
+  git clone ${SRC_URL}
```
Comment (starwarfan) [inline scripts/release/install_ffmpeg.sh:107]: Better use a certain version after clone

File: source/agent/streaming/index.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
     var internalConnFactory = new InternalConnectionFactory;
 
     var notifyStatus = (controller, sessionId, direction, status) => {
+	log.info('notify status:', controller, ' sessionId:', sessionId, ' direction:', direction, ' status:',status);
```
Comment (starwarfan) [inline source/agent/streaming/index.js:32]: indent and use debug here

File: scripts/installCommonDeps.sh (hunk 1 above)
Comment (qwu16) [reply scripts/installCommonDeps.sh]: Updated

File: scripts/release/init-all.sh (hunk 2 above)
Comment (qwu16) [reply scripts/release/init-all.sh:42]: Yes, it is mandatory in streaming agent for SRT

File: scripts/release/install_ffmpeg.sh (hunk 3 above)
Comment (qwu16) [reply scripts/release/install_ffmpeg.sh]: updated

File: source/agent/streaming/index.js (hunk 4 above)
Comment (qwu16) [reply source/agent/streaming/index.js]: updated

File: source/agent/streaming/index.js [hunk 5]
```diff
@@ -120,25 +147,119 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
...
         switch (connectionType) {
             case 'internal':
                 conn = internalConnFactory.fetch(connectionId, 'in');
                 if (conn)
                     conn.connect(options);
                 break;
             case 'streaming':
-                conn = createAVStreamIn(connectionId, options);
-                break;
+		      if(options.connection.type == 'srt') {
```
Comment (starwarfan) [inline source/agent/streaming/index.js:158]: incorrect indent

File: source/agent/streaming/index.js [hunk 6]
```diff
@@ -120,25 +147,119 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
...
+                        if(options.connection.listen_timeout) {
+                            srturl = srturl + '&listen_timeout=' + options.connection.listen_timeout;
+                        }
+
+                        var srtoptions = {
+                            controller: options.controller,
+                            connection: {
+                                url:srturl,
+                                transportProtocol:'udp'
+			                 },
```
Comment (starwarfan) [inline source/agent/streaming/index.js:185]: indent

File: source/agent/streaming/index.js [hunk 7]
```diff
@@ -120,25 +147,119 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
...
+                        controller: options.controller,
+                        connection: {
+                            url:srturl,
+                            transportProtocol:'udp'
+                         },
+                        media: options.media};
+                        srtresult = {
+                                streamid: connectionId
+                            };
+                        callback('callback', srtresult);
```
Comment (starwarfan) [inline source/agent/streaming/index.js:234]: Here 'createAVStreamIn' is not called

File: source/agent/streaming/index.js [hunk 8]
```diff
@@ -120,25 +147,119 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
...
+                        }
+
+                        srtresult = {
+                            ip : address,
+                            port : port,
+                            streamid: connectionId
+                        };
+
+                        log.debug("srt result is:", srtresult);
+			             log.debug('srt options is:', srtoptions);
```
Comment (starwarfan) [inline source/agent/streaming/index.js:204]: Bad indent, you can use some tools like js beautify to format your code before submit.

**Review by daijh — APPROVED** [review]

**Pull request merged**